pkg/cmd/diff/diff.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2024 The KitOps Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package diff
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"oras.land/oras-go/v2/registry"
+
+	"kitops/pkg/lib/constants"
+	"kitops/pkg/lib/repo/local"
+	"kitops/pkg/lib/repo/remote"
+	"kitops/pkg/lib/repo/util"
+)
+
+// Helper struct diffInfo holds the manifest and its descriptor for a ModelKit.
+type diffInfo struct {
+	Manifest   *ocispec.Manifest
+	Descriptor ocispec.Descriptor
+}
+
+// Helper struct diffResult contains the comparison results between two ModelKits.
+type diffResult struct {
+	SharedLayers  []ocispec.Descriptor
+	UniqueLayersA []ocispec.Descriptor
+	UniqueLayersB []ocispec.Descriptor
+}
+
+// compareManifests compares two OCI manifests and returns the shared and unique layers.
+func compareManifests(manifestA *ocispec.Manifest, manifestB *ocispec.Manifest) *diffResult {
+	result := &diffResult{}
+
+	layerMapA := make(map[string]ocispec.Descriptor)
+	for _, layer := range manifestA.Layers {
+		layerMapA[layer.Digest.String()] = layer
+	}
+
+	for _, layer := range manifestB.Layers {
+		if _, ok := layerMapA[layer.Digest.String()]; ok {
+			result.SharedLayers = append(result.SharedLayers, layer)
+			delete(layerMapA, layer.Digest.String())
+		} else {
+			result.UniqueLayersB = append(result.UniqueLayersB, layer)
+		}
+	}
+
+	result.UniqueLayersA = make([]ocispec.Descriptor, 0, len(layerMapA))
+	for _, layer := range layerMapA {
+		result.UniqueLayersA = append(result.UniqueLayersA, layer)
+	}
```

Comment:
We might want to impose some order on these layers; they'll be iterated randomly from the map by Go.